src/pages/api/profile/user-profile/index.ts

```diff
@@ -168,8 +168,8 @@ const postHandler: Handler = async (request, response) => {
         true,
       );
 
-      response.statusCode = HttpStatus.OK;
-      response.json({ createdUserProfile });
+      response.statusCode = HttpStatus.CREATED;
```

Comment:
And as for address, we don't rely on `createdUserProfile` or `updatedUserProfile` keys anymore for the redirection part but now we use response code: `200` for the update and `201` for the create.